File: system/src/main/resources/static/system/js/service/goodsService.js
//服务层
app.service('goodsService',function($http){

	//分页 
	this.findPage=function(page,rows){
		return $http.get(basePath+"/goods/findPage?page="+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get(basePath+"/goods/findOne?id="+id);
	}
	//审核
	this.changeStatus=function(id,status,examine_errorMsg){
		return $http.put(basePath+"/goods/examine?id="+id+'&status='+status+'&examineErrorMsg='+examine_errorMsg);
	}
	//批量审核
	this.changeStatusSelected=function(ids,status,examine_errorMsg){
		return $http.put(basePath+"/goods/examineSelected?ids="+ids+'&status='+status+'&examineErrorMsg='+examine_errorMsg);
	}
	//增加 
	this.add=function(entity){
		return  $http.post(basePath+"/goods/add",entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post(basePath+"/goods/update",entity );
	}
	//商家更新商品状态
	this.updateStatus=function(ids,status){
		return $http.put(basePath+"/goods/updateStatus?ids="+ids+"&status="+status);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post(basePath+"/goods/search?page="+page+"&rows="+rows, searchEntity);
	}
	//初始化已上架商品搜索引擎
	this.importGoods=function(p){
		return $http.get(basePath+"/goods_engine/importGoods");
	}
});

File: system/src/main/java/com/cys/system/common/service/GoodsService.java
package com.cys.system.common.service;

import com.cys.system.common.common.pojo.Result;
import com.cys.system.common.pojo.Goods;
import com.cys.system.common.pojo.Product;

import java.util.List;
import java.util.Map;

public interface GoodsService {
    Result listGoods(Integer page, Integer rows, Goods goods);

    Result findGoodsById(Integer id);

    Result examine(Integer id, Integer status);

    Result examineSelected(Integer[] ids, Integer status);

    Result addGoods(Goods goods, Integer shopId);

    Result updateGoodsByGoods(Goods goods);

    void deleteGoodsById(Integer id);

    Result updateStatus(Integer[] ids, Integer status);

    Product saleProduct();

    Result transactionAnalysis();

    List<Map> listDeleteGoods();

    Result findGoodGoods();

    Result recomment(Integer id,Integer status);
}
